easybuild/tools/module_generator.py

```diff
@@ -238,18 +285,181 @@ def set_alias(self, key, value):
         # quotes are needed, to ensure smooth working of EBDEVEL* modulefiles
         return 'set-alias\t%s\t\t%s\n' % (key, quote_str(value))
 
-    def set_fake(self, fake):
-        """Determine whether this ModuleGenerator instance should generate fake modules."""
-        _log.debug("Updating fake for this ModuleGenerator instance to %s (was %s)" % (fake, self.fake))
-        self.fake = fake
-        # fake mode: set installpath to temporary dir
-        if self.fake:
-            self.tmpdir = tempfile.mkdtemp()
-            _log.debug("Fake mode: using %s (instead of %s)" % (self.tmpdir, self.module_path))
-            self.module_path = self.tmpdir
+
+class ModuleGeneratorLua(ModuleGenerator):
+    """
+    Class for generating Lua module files.
+    """
+    SYNTAX = LUA_SYNTAX
+
+    MODULE_SUFFIX = '.lua'
+
+    def __init__(self, *args, **kwargs):
+        """ModuleGeneratorLua constructor."""
+        super(ModuleGeneratorLua, self).__init__(*args, **kwargs)
+
+    def module_header(self):
+        """Return module header string."""
+        return ''
+
+    def comment(self, msg):
+        """Return string containing given message as a comment."""
+        return " -- %s\n" % msg
+
+    def get_description(self, conflict=True):
+        """
+        Generate a description.
+        """
+
+        description = "%s - Homepage: %s" % (self.app.cfg['description'], self.app.cfg['homepage'])
+
+        lines = [
+            "local pkg = {}",
+            "help = [["
+            "%(description)s"
+            "]]",
+            "whatis([[Name: %(name)s]])",
+            "whatis([[Version: %(version)s]])",
+            "whatis([[Description: %(description)s]])",
+            "whatis([[Homepage: %(homepage)s]])"
+            "whatis([[License: N/A ]])",
+            "whatis([[Keywords: Not set]])",
+            "",
+            "",
+            'pkg.root="%(installdir)s"',
+            "",
+            ]
+
+        #@todo check if this is really needed, imho Lmod doesnt need this at all.
+        if self.app.cfg['moduleloadnoconflict']:
+            lines.extend([
+             'if ( not isloaded("%(name)s/%(version)s")) then',
+             '  load("%(name)s/%(version)s")',
+             'end',
+             ])
+
+        elif conflict:
```

Comment:
The conflict on the same module name doesn't work but we might want to add a conflict on other modules. Say you have a conflict between Oracle JRE and OpenJDK.

Especially for bioinformatics packages we might want to add a conflict on samtools, bwa and other common binaries that are sometimes shipped as part of the package and we might want to add conflict support here to prevent surprising side effects by having a EB provided samtools shadow an included samtools for example.

 We dont have an API to expose conflict on other modules yet, but we want to add this at some point in time.